integrations/amazon_bedrock/src/haystack_integrations/components/downloaders/s3/s3_downloader.py

```diff
@@ -0,0 +1,405 @@
+        download_dir=str(Path.cwd() / "s3_cache"),
+        file_extensions=[".json"],
+        sources_target_type="str",
+    )
+
+    docs = [
+        Document(content="", meta={"s3_bucket": "bucket_name", "s3_key": "your_key", "file_name": "file_name"}),
+    ]
+    result = downloader.run(documents=docs)
+    print("Downloaded docs:")
+    for d in result["documents"]:
+        print(f"- {d.meta.get('file_name')} -> {d.meta.get('file_path')}")
+
+    ```
+    """
+
+    max_cache_size: int = 100
+
+    def __init__(
+        self,
+        aws_access_key_id: Optional[Secret] = Secret.from_env_var("AWS_ACCESS_KEY_ID", strict=False),  # noqa: B008
+        aws_secret_access_key: Optional[Secret] = Secret.from_env_var(  # noqa: B008
+            "AWS_SECRET_ACCESS_KEY", strict=False
+        ),
+        aws_session_token: Optional[Secret] = Secret.from_env_var("AWS_SESSION_TOKEN", strict=False),  # noqa: B008
+        aws_region_name: Optional[Secret] = Secret.from_env_var("AWS_DEFAULT_REGION", strict=False),  # noqa: B008
+        aws_profile_name: Optional[Secret] = Secret.from_env_var("AWS_PROFILE", strict=False),  # noqa: B008
+        verify_ssl: bool = True,
+        boto3_config: Optional[Dict[str, Any]] = None,
+        download_dir: Optional[str] = None,
+        file_extensions: Optional[List[str]] = None,
+        sources_target_type: Literal["str", "pathlib.Path", "haystack.dataclasses.ByteStream"] = "str",
+        max_workers: int = 32,
+        max_cache_size: int = 100,
+    ) -> None:
+        """
+        Initializes the S3Downloader with the provided parameters. The parameters are passed to the
+        Amazon S3 client.
+
+        :param aws_access_key_id: AWS access key ID.
+        :param aws_secret_access_key: AWS secret access key.
+        :param aws_session_token: AWS session token.
+        :param aws_region_name: AWS region name.
+        :param aws_profile_name: AWS profile name.
+        :param verify_ssl: Whether to verify SSL.
+        :param boto3_config: The configuration for the boto3 client.
+        :param download_dir: The directory where the files will be downloaded.
+        :param file_extensions: The file extensions that are permitted to be downloaded.
+        :param sources_target_type: The target type for the sources.
+            Can be "str", "pathlib.Path" or
+            "haystack.dataclasses.ByteStream".
+        :param max_workers: The maximum number of workers to use for concurrent downloads.
+        :param max_cache_size: The maximum number of files to cache.
+        :raises AmazonS3ConfigurationError: If the AWS configuration is not correct.
+        """
+
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.aws_region_name = aws_region_name
+        self.aws_session_token = aws_session_token
+        self.aws_profile_name = aws_profile_name
+        self.verify_ssl = verify_ssl
```

Comment:
I wasn’t able to find where `verify_ssl` is being used in the code. Could you point me to where it’s referenced or explain how it’s intended to be used?